app/assets/javascripts/specifics.js.coffee

```diff
@@ -564,7 +569,23 @@ class Row
   toHashKey: ->
     @flattenToIds().join(",") + ",#{@specificLeftMost}" + ",#{@nonSpecificLeftMost?.id}"
 
-  
+  # If the row references a leftmost, either specific or not, return the event(s)
+  # (because a UNION can place multiple events in the specific leftMost, this can be > 1)
+  leftMostEvents: ->
+    if @nonSpecificLeftMost?
+      return [@nonSpecificLeftMost]
+    if @specificLeftMost? && _.isString(@specificLeftMost)
+      specificIndex = hqmf.SpecificsManager.getColumnIndex(@specificLeftMost)
+      return [@values[specificIndex]] if @values[specificIndex]? && @values[specificIndex] != hqmf.SpecificsManager.any
+    if @specificLeftMost? && _.isObject(@specificLeftMost)
+      events = []
```

Comment:
I think that there can be repeated events; while I don't think they would necessarily be an issue (the previous approach was only a partial solution to repeated events anyway), I've tweaked the code to more closely match the previous approach and re-run the regression tests.